.github/workflows/ci.yaml

```diff
@@ -2,62 +2,30 @@ name: CI
-  # sometimes happen when (for example) the OS version is changed but
-  # older .so files are cached, which can have various effects
-  # (e.g. cabal complains it can't find a valid version of the "happy"
-  # tool).
-  CACHE_VERSION: 1
-
 jobs:
-  build-linux:
-    runs-on: ubuntu-20.04
+  build:
+    name: GHC v${{ matrix.ghc }}
     strategy:
       matrix:
-        os: [ubuntu-22.04]
-        ghc-ver: ["9.6.4", "9.8.1", "9.10.1"]
-        cabal: ["3.10.2.0"]
-      # complete all jobs
+        cabal: ["3.14.2.0"]
+        ghc: ["9.6.4", "9.8.1", "9.10.1"]
+        os: ["ubuntu-24.04"]
       fail-fast: false
-    name: GHC v${{ matrix.ghc-ver }}
-    steps:
-    - name: Checkout
-      uses: actions/checkout@v2
-      with:
-        submodules: true
-    - name: Copy cabal project files
-      run: |
+    uses: GaloisInc/.github/.github/workflows/haskell-ci.yml@v1
+    with:
+      cabal: ${{ matrix.cabal }}
+      # The numeric prefix can be updated to force the use of a new cache if
+      # the current cache contents become corrupted/invalid.  This can
+      # sometimes happen when (for example) the OS version is changed but
+      # older .so files are cached, which can have various effects
+      # (e.g. cabal complains it can't find a valid version of the "happy"
+      # tool).
+      cache-key-prefix: 1
+      check: false
+      ghc: ${{ matrix.ghc }}
+      os: ${{ matrix.os }}
+      submodules: "true"
+      pre-hook: |
         cp cabal.project.dist cabal.project
-        cp cabal.project.werror cabal.project.local
-        cp cabal.project.ghc-${{ matrix.ghc-ver }}.freeze cabal.project.freeze
-    - uses: haskell/actions/setup@v1
-      id: setup-haskell
-      name: Setup Haskell
-      with:
-        ghc-version: ${{ matrix.ghc-ver }}
-        cabal-version: ${{ matrix.cabal }}
-    - uses: actions/cache/restore@v3
-      name: Restore cabal store cache
-      with:
-        path: |
-          ${{ steps.setup-haskell.outputs.cabal-store }}
-          dist-newstyle
-        key: ${{ env.CACHE_VERSION }}-cabal-${{ matrix.os }}-${{ matrix.ghc-ver }}-${{ hashFiles(format('cabal.project.ghc-{0}.freeze', matrix.ghc-ver)) }}-${{ github.sha }}
-        restore-keys: |
-          ${{ env.CACHE_VERSION }}-cabal-${{ matrix.os }}-${{ matrix.ghc-ver }}-${{ hashFiles(format('cabal.project.ghc-{0}.freeze', matrix.ghc-ver)) }}-
-    - name: Cabal update
-      run: cabal update
-    - name: Build
-      run: cabal build
-    - name: Test
-      run: cabal test
-    - uses: actions/cache/save@v3
-      name: Save cabal store cache
-      if: always()
-      with:
-        path: |
-          ${{ steps.setup-haskell.outputs.cabal-store }}
-          dist-newstyle
-        key: ${{ env.CACHE_VERSION }}-cabal-${{ matrix.os }}-${{ matrix.ghc-ver }}-${{ hashFiles(format('cabal.project.ghc-{0}.freeze', matrix.ghc-ver)) }}-${{ github.sha }}
+        cat cabal.project.werror >> cabal.project.local
```

Comment:
Why not just `cp cabal.project.werror cabal.project.local` as before?

Reply:
Oh, I guess I was thinking that `cabal configure` creates `cabal.project.local`, but now that I think about it for a second that happens after the `pre-hook`. Will change back to `cp`.

Reply:
Oh wait, this is a little bit of a problem:
```
'cabal.project.local' already exists, backing it up to 'cabal.project.local~'.
```
But the `.werror` file specifies GHC options for a single package, which AFAIK can't be done through `cabal configure`... Could add a `post-configure-hook` and do the `cat >>` there, but that seems a bit inelegant...

Ideally we'd just use `-Werror` for all local packages, but that currently chokes on `elf-edit`. Perhaps I'll just fix that warning and enable `-Werror` upstream and do the same here.